examples/preprocessing/plot_target_regressor_encoder.py

```diff
@@ -0,0 +1,121 @@
+"""
+=============================
+Target Encoder for Regressors
+=============================
+
+.. currentmodule:: sklearn.preprocessing
+
+The :class:`TargetRegressorEncoder` uses target statistics conditioned on
+the categorical features for encoding. In this example, we will compare
+:class:`TargetRegressorEncoder`, :class:`OrdinalEncoder`, and dropping the
+category on a wine review dataset.
+"""
+
+# %%
+# Loading Data from OpenML
+# ========================
+# First, we load the wine reviews dataset, where the target is the points given
+# be a reviewer:
+from sklearn.datasets import fetch_openml
+
+wine_reviews = fetch_openml(data_id=42074, as_frame=True, parser="liac-arff")
+
+df = wine_reviews.frame
+df.head()
+
+# %%
+# For this example, we use the following subset of numerical and categorical
+# features in the data. The categorical features have a cardinality ranging
+# from 18 to 14810:
+numerical_features = ["price"]
+categorical_features = [
+    "country",
+    "province",
+    "region_1",
+    "region_2",
+    "variety",
+    "winery",
+]
+
+X = df[numerical_features + categorical_features]
+y = df["points"]
+X.nunique().sort_values(ascending=False)
+
+# %%
+# We split the dataset into a training and test set:
+from sklearn.model_selection import train_test_split
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)
+
+print(f"Samples in training set: {len(X_train)}\nSamples in test set: {len(X_test)}")
+
+# %%
+# Building and Training Pipelines with Different Encoders
+# =======================================================
+# Dropping the categorical features
+# ---------------------------------
+# As a basline, we construct a pipeline where the categorical features are
+# dropped.
+from sklearn.compose import ColumnTransformer
+from sklearn.pipeline import Pipeline
+from sklearn.ensemble import HistGradientBoostingRegressor
+
+prep = ColumnTransformer(
+    [
+        ("num", "passthrough", numerical_features),
+        ("cat", "drop", categorical_features),
+    ]
+)
+
+reg_drop_cats = Pipeline(
+    [("prep", prep), ("hist", HistGradientBoostingRegressor(random_state=0))]
+)
+reg_drop_cats
+
+# %%
+# Here we train and use the root mean squared error to evalute the baseline
+# model:
```

Comment:
Is there a short sentence for why RMSE is a good/sensible choice? I've not looked at the data but I could imagine that the score/points/rating the reviewers assign are like `1`, `2`, `3`, etc, not truly continuous values. As a reader I'm now wondering why it is called out that the obvious(?) choice of RMSE is used for evaluating the performance of the pipeline, but we didn't talk about the default parameters of the gradient boosting model.

So if there is a nice short sentence to explain it, that would be great. Otherwise, I understand we need some kind of sentence for this cell :-/ but it would be nice to use it to add to what the code already says (ideally explaining why the code is what it is).

Reply:
The unique points are:

```python
array([ 96.,  95.,  94.,  90.,  91.,  86.,  89.,  88.,  87.,  93.,  92.,
        85.,  84.,  83.,  82.,  81., 100.,  99.,  98.,  97.,  80.])
```

I wanted to model this as regression, because there is an intrinsic order to the points. The other way to keep the order semantics would be Ordinal Regression, but I do not think anyone has look into it for target encoders.

I agree, this section can be expanded a little more.

**Aside**: [Here is a fun library](https://raschka-research-group.github.io/coral-pytorch/) that does Ordinal Regression.

Reply:
I removed the mention of RMSE. My aim is to focus the narrative on encoding and less on the choice of metric. Choosing a metric for evaluation is a much bigger topic and I do not want to squeeze it into one sentence.

Reply:
Sounds good.